src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,168 @@
+package duke;
+
+import duke.command.*;
+import duke.exceptions.InvalidFileTypeException;
+import duke.exceptions.InvalidTaskException;
+import duke.task.Deadlines;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Used for any parsing of user input
+ */
+public class Parser {
+    public Parser() {}
+
+    public Matcher regexParse(String regex, String text) {
+        Pattern pattern = Pattern.compile(regex);
+        Matcher matcher = pattern.matcher(text);
+        return matcher;
+    }
+
+    /**
+     * Parses the instruction that has been typed in by a user, returning the command object to be executed
+     * @param response the instruction typed in by the user
+     * @return the command object to be executed
+     * @throws InvalidTaskException
+     */
+    public Command parse(String response) throws InvalidTaskException {
+        int taskIndex;
+        Matcher matcher;
+
+        switch(response.split(" ")[0]) {
+            case "list":
+                return new PrintListCommand();
+            case "mark":
+            case "unmark":
+                boolean done = response.split(" ")[0].equals("mark") ? true : false;
+                taskIndex = Integer.parseInt(response.split(" ")[1]) - 1;
+                return new MarkCommand(done, taskIndex);
+            case "todo":
+                matcher = this.regexParse("^todo\\s([\\w\\s]*)$", response);
+                if (!matcher.find() || matcher.groupCount() != 1) {
+                    throw new InvalidTaskException("Invalid use of todo. Usage: todo <task description>");
+                }
+                return new ToDoCommand(matcher.group(1), false);
+            case "deadline":
+                matcher = regexParse("^deadline\\s([\\w\\s]*)\\s\\/by\\s([\\w\\s\\:\\-]*)$", response);
+                if (!matcher.find() || matcher.groupCount() != 2) {
+                    throw new InvalidTaskException(
+                            "Invalid use of deadline. Usage: deadline <task description> /by <date & time>"
+                    );
+                }
+                return new DeadlineCommand(matcher.group(1), matcher.group(2), false);
```

Comment:
Do the statements in the case blocks follow the Single level of abstraction principle?